### DISCUSSION TITLE
In config, is it possible to serialize an object to JSON so I can use it as an env. variable?

### DISCUSSION THREAD

**Discussion by ronkorving** [serverless/serverless#10312] [Q&A]
I want to send some fancy JSON as an environment variable to my Lambda. The source of that is part of my Serverless configuration under "custom". Is there a way I can refer to that, serialize it to JSON and use it as an environment variable for my Lambda? If not, does anyone know of a plugin that can do this?

**Comment by pgrzesik** [comment]
Hey @ronkorving - I don't know of such a plugin/tool, but you can get an inspiration to write a simple plugin of your own from this one: https://github.com/whardier/serverless-plugin-powertools

**Comment by ronkorving** [comment]
@pgrzesik Thanks, I noticed that plugin too, and considered exactly what you're saying. In the end, I ended up writing a `${file(serverless-complex-var.js):myKey` file that looks like this:

```js
// Returns a base64 string of a JSON serialized value
// Useful when trying to inject a piece of configuration as JSON as an environment variable to a Lambda

const resolve = (serverless, path) => {
  // This is the legacy resolver, which we need to use, because the serverless-plugin-function-value plugin
  // still depends on this mode. When that changes, we can switch to the improved async method.
  // See: https://www.serverless.com/framework/docs/providers/aws/guide/variables#exporting-a-function

  let complexVar = serverless.configurationInput;

  for (const step of path) {
    complexVar = complexVar[step];

    if (complexVar === undefined) {
      return '';
    }
  }

  return Buffer.from(JSON.stringify(complexVar)).toString('base64');
};

exports.myKey = (serverless) => {
  return resolve(serverless, ['custom', 'config', 'myKey']);
};
```

It's not awesome, because you need to set all keys, possibly in vain, but it works, and for now, I only have one key.

**Reply by pgrzesik** [reply]
That's also a valid way to go about it 👍
